src/html/js/app.js

```diff
@@ -250,6 +284,177 @@ createInfoTd = function(i, accountInfo) {
+       var out_html = $($.parseHTML(data));
+       $("#loadContent").empty();
+       let id = PCalendar.createSubContent(out_html);
+       $(id + " main").empty();
+       $(id + " .header-title").attr("data-i18n", "glossary:Sharing.Received.label").localize();
+       $(id + " footer").hide();
+       $(id + " .header-btn-right").hide();
+       PCalendar.getReceivedMessageAPI().done(function(data) {
+           dispReceivedSharingList(id + " main", data);
+       });
+   }).fail(function(error) {
+       console.log(error);
+   });
+}
+
+dispReceivedSharingList = function(id, results) {
+    let eUl = $('<ul>', {
+        class: 'slide-list hover-action',
+        id: 'sendSharingCells'
+    });
+    $(id).append($(eUl)).localize();
+    let res = results.d.results;
+    res.sort(function(val1, val2) {
+        return (val1.Url < val2.Url ? 1 : -1);
+    })
+    reqReceivedUUID = {};
+    reqRequestAuthority = {};
+    for (var i in res) {
+        if (res[i].Status == "approved" || res[i].Status == "rejected" || !res[i].RequestObjects) {
+            continue;
+        }
+
+        reqReceivedUUID[i] = res[i].__id;
+        reqRequestAuthority[i] = res[i].RequestObjects[0].Name;
+        let extUrl = Common.changeLocalUnitToUnitUrl(res[i].From);
+
+        Common.getProfileName(extUrl, Common.prepareExtCellForApp, i);
+    }
+    if (Object.keys(reqReceivedUUID).length == 0) {
+        $('body > div.mySpinner').hide();
+    }
+}
+
+dispReceivedCellInfo = function(extUrl, uuid, eleId, reqAuth) {
+    Common.loadContent("./templates/_shared_template.html").done(function(data) {
+        var out_html = $($.parseHTML(data));
+        let id = PCalendar.createSubContent(out_html);
+        $(id + " .singleBtn").hide();
+        $(id + " .header-btn-right").hide();
+        $(id + " .sendShared").hide();
+        $(id + " #reqAuthority").attr("data-i18n", "glossary:Sharing.Authority." + reqAuth).localize();
+        $(id + " #approvalFooterButton").removeAttr("onclick").on("click", function() {
+            Common.approvalRel(extUrl, uuid, eleId, PCalendar.backSubContent);
+        });
+        $(id + " #refectionFooterButton").removeAttr("onclick").on("click", function() {
+            Common.rejectionRel(extUrl, uuid, eleId, PCalendar.backSubContent);
+        });
+        Common.getProfileName(extUrl, function(url, profObj) {
+            $(id + " .header-title").text(profObj.dispName);
+            $(id + " .user-cell-url").text(url);
+            $(id + " .user-description").text(profObj.description);
+            $(id + " .extcell-profile .user-icon").append('<img class="user-icon-large" src="' + profObj.dispImage + '" alt="user">');
+        });
+    }).fail(function(error) {
+        console.log(error);
+    });
+}
+
+/*
+ * Get Transcell Token of the external Cell and prepare its data.
+ * When done, execute callback (add external Cell to proper list).
+ */
+Common.prepareExtCellForApp = function(extUrl, profObj, no) {
```

Comment:
Please don't overwrite existing method.
Similar method is already defined in common_personium.js.

Reply:
It has been deleted

Reply:
Did you modify anything to fit your purpose?

Reply:
We are also updating Common.prepareExtCellForApp of common_personium.js